server/src/main/java/io/crate/execution/engine/join/HashJoinOperation.java

```diff
@@ -57,7 +57,8 @@ public HashJoinOperation(int numLeftCols,
                              TransactionContext txnCtx,
                              InputFactory inputFactory,
                              CircuitBreaker circuitBreaker,
-                             long estimatedRowSizeForLeft) {
+                             long estimatedRowSizeForLeft,
+                             boolean emitNullValues) {
```

Comment:
Maybe we can already have an enum like:
```
enum EmitNulls {
    LEFT,
    RIGHT,
    BOTH
}
```
`null` means no, or we can have another value in the enum: `NONE`
 so we're preparing the ground for the other outer joins.
It can have a method that converts the JoinType into EmitNulls, or we can pass down to the iterator the JoinType itself and the iterator makes the translation.

Reply:
Right we would `NONE` and `LEFT` and eventually `BOTH` if we decide to implement a full-outer. I think with two states i am ok with boolean.